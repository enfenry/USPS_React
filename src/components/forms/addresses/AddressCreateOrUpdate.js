import React from 'react';
import { Button, Form, FormGroup } from 'reactstrap';
import { reduxForm, Field } from 'redux-form';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import validate from './AddressValidate';
import renderField from '../formUtils';

let AddressCreateOrUpdate = props => {
  const { handleSubmit, handleHide } = props;

  return (
    <Form onSubmit={handleSubmit} className="form">
      <FormGroup className="field">
        <div className="control">
          <Field name="ss_name" component={renderField} type="text" label="Name" />
        </div>
      </FormGroup>

      <FormGroup className="field">
        <div className="control">
          <Field name="ss_line1" component={renderField} type="text" label="Line 1" />
        </div>
      </FormGroup>

      <FormGroup className="field">
        <div className="control">
          <Field name="ss_line2" component={renderField} type="text" label="Line 2" />
        </div>
      </FormGroup>

      <FormGroup className="field">
        <div className="control">
          <Field name="ss_line3" component={renderField} type="text" label="Line 3" />
        </div>
      </FormGroup>

      <FormGroup className="field">
        <div className="control">
          <Field name="ss_city" component={renderField} type="text" label="City" />
        </div>
      </FormGroup>

      <FormGroup className="field">
        <div className="control">
          <Field name="ss_state" component={renderField} type="text" label="State" />
        </div>
      </FormGroup>

      <FormGroup className="field">
        <div className="control">
          <Field name="ss_country" component={renderField} type="text" label="Country" />
        </div>
      </FormGroup>

      <FormGroup className="field">
        <div className="control">
          <Field name="ss_postalcode" component={renderField} type="text" label="Postal Code" />
        </div>
      </FormGroup>

      <div className="field">
        <div className="control">
          <Button className="button is-link" color="success" onClick={handleSubmit}>Submit</Button>{' '}
          <Button color="secondary" onClick={handleHide}>Cancel</Button>
        </div>
      </div>
    </Form>
  );
};

AddressCreateOrUpdate.propTypes = {
  handleSubmit: PropTypes.func,
  handleHide: PropTypes.func,
  initialValues: PropTypes.object,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
}

AddressCreateOrUpdate = reduxForm({
  form: 'AddressCreateOrUpdate',
  validate,
  onSubmitSuccess: (result, dispatch, props) => {
    props.handleHide();
  }
})(AddressCreateOrUpdate);

export default connect()(AddressCreateOrUpdate);