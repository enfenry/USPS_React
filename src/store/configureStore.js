import { createStore, applyMiddleware, compose } from 'redux';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
import thunk from 'redux-thunk';
import rootReducer from '../reducers';

import { createLogger } from 'redux-logger'
import DevTools from '../containers/DevTools';

function configureStore(initialState) {

  const logger = createLogger()

  const middlewares = [
    reduxImmutableStateInvariant(),
    thunk,
    logger
  ];

  const store = createStore(
    rootReducer,
    initialState,
    compose(
      applyMiddleware(...middlewares),
      DevTools.instrument()
    )
  );

  return store;
}

export default configureStore;
