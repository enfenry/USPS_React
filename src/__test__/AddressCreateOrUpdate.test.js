import React from 'react';
import AddressCreateOrUpdate from '../components/forms/addresses/AddressCreateOrUpdate';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import basicState from "./basicState";

import configureStore from 'redux-mock-store';
const mockStore = configureStore([]);

test('AddressCreateOrUpdate renders correctly', () => {

    const store = mockStore(basicState);
    const initialValues = basicState.addressesReducer.addresses[0];

    const component = renderer.create(
        <Provider store={store}>
            <AddressCreateOrUpdate initialValues={initialValues} />
        </Provider>
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});