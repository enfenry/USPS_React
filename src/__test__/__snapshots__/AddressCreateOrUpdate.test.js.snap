// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressCreateOrUpdate renders correctly 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <div
    className="field form-group"
  >
    <div
      className="control"
    >
      <div
        style={
          Object {
            "display": "block",
          }
        }
      >
        <div
          className="control"
        >
          <label
            className="field"
          >
            Name
          </label>
          <input
            className="input form-control"
            name="ss_name"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="field form-group"
  >
    <div
      className="control"
    >
      <div
        style={
          Object {
            "display": "block",
          }
        }
      >
        <div
          className="control"
        >
          <label
            className="field"
          >
            Line 1
          </label>
          <input
            className="input form-control"
            name="ss_line1"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="field form-group"
  >
    <div
      className="control"
    >
      <div
        style={
          Object {
            "display": "block",
          }
        }
      >
        <div
          className="control"
        >
          <label
            className="field"
          >
            Line 2
          </label>
          <input
            className="input form-control"
            name="ss_line2"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="field form-group"
  >
    <div
      className="control"
    >
      <div
        style={
          Object {
            "display": "block",
          }
        }
      >
        <div
          className="control"
        >
          <label
            className="field"
          >
            Line 3
          </label>
          <input
            className="input form-control"
            name="ss_line3"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="field form-group"
  >
    <div
      className="control"
    >
      <div
        style={
          Object {
            "display": "block",
          }
        }
      >
        <div
          className="control"
        >
          <label
            className="field"
          >
            City
          </label>
          <input
            className="input form-control"
            name="ss_city"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="field form-group"
  >
    <div
      className="control"
    >
      <div
        style={
          Object {
            "display": "block",
          }
        }
      >
        <div
          className="control"
        >
          <label
            className="field"
          >
            State
          </label>
          <input
            className="input form-control"
            name="ss_state"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="field form-group"
  >
    <div
      className="control"
    >
      <div
        style={
          Object {
            "display": "block",
          }
        }
      >
        <div
          className="control"
        >
          <label
            className="field"
          >
            Country
          </label>
          <input
            className="input form-control"
            name="ss_country"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="field form-group"
  >
    <div
      className="control"
    >
      <div
        style={
          Object {
            "display": "block",
          }
        }
      >
        <div
          className="control"
        >
          <label
            className="field"
          >
            Postal Code
          </label>
          <input
            className="input form-control"
            name="ss_postalcode"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="field"
  >
    <div
      className="control"
    >
      <button
        aria-label={null}
        className="button is-link btn btn-success"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
       
      <button
        aria-label={null}
        className="btn btn-secondary"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
</form>
`;
